src/Microsoft.ML.Transforms/Text/TextNormalizerTransform.cs

```diff
@@ -215,171 +156,426 @@ public override void Save(ModelSaveContext ctx)
+            SaveColumns(ctx);
+            foreach (var column in _columns)
+            {
+                ctx.Writer.Write((byte)column.TextCase);
+                ctx.Writer.WriteBoolByte(column.KeepDiacritics);
+                ctx.Writer.WriteBoolByte(column.KeepPunctuations);
+                ctx.Writer.WriteBoolByte(column.KeepNumbers);
+            }
         }
 
-        protected override ColumnType GetColumnTypeCore(int iinfo)
+        // Factory method for SignatureLoadModel.
+        private static TextNormalizerTransform Create(IHostEnvironment env, ModelLoadContext ctx)
         {
-            Host.Assert(0 <= iinfo & iinfo < Infos.Length);
-            return Infos[iinfo].TypeSrc.IsVector ? new VectorType(TextType.Instance) : Infos[iinfo].TypeSrc;
+            Contracts.CheckValue(env, nameof(env));
+            var host = env.Register(RegistrationName);
+            host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel(GetVersionInfo());
+            return new TextNormalizerTransform(host, ctx);
         }
 
-        protected override Delegate GetGetterCore(IChannel ch, IRow input, int iinfo, out Action disposer)
+        private TextNormalizerTransform(IHost host, ModelLoadContext ctx)
+          : base(host, ctx)
         {
-            Host.AssertValueOrNull(ch);
-            Host.AssertValue(input);
-            Host.Assert(0 <= iinfo && iinfo < Infos.Length);
-            disposer = null;
-
-            var typeSrc = Infos[iinfo].TypeSrc;
-            Host.Assert(typeSrc.ItemType.IsText);
-
-            if (typeSrc.IsVector)
+            var columnsLength = ColumnPairs.Length;
+            if (ctx.Header.ModelVerWritten == 0x00010001)
             {
-                Host.Assert(typeSrc.VectorSize >= 0);
-                return MakeGetterVec(input, iinfo);
+                // *** Binary format ***
+                // <base>
+                //   byte: case
+                //   bool: whether to keep diacritics
+                //   bool: whether to keep punctuations
+                //   bool: whether to keep numbers
+                var textCase = (TextNormalizerEstimator.CaseNormalizationMode)ctx.Reader.ReadByte();
+                host.CheckDecode(Enum.IsDefined(typeof(TextNormalizerEstimator.CaseNormalizationMode), textCase));
+
+                var keepDiacritics = ctx.Reader.ReadBoolByte();
+                var keepPunctuations = ctx.Reader.ReadBoolByte();
+                var keepNumbers = ctx.Reader.ReadBoolByte();
+                _columns = new ColumnInfo[columnsLength];
+                for (int i = 0; i < columnsLength; i++)
+                    _columns[i] = new ColumnInfo(ColumnPairs[i].input, ColumnPairs[i].output, textCase, keepDiacritics, keepPunctuations, keepNumbers);
+            }
+            else
+            {
+                _columns = new ColumnInfo[columnsLength];
+                for (int i = 0; i < columnsLength; i++)
+                {
+                    // *** Binary format ***
```

Comment:
>/ *** Binary format *** [](start = 21, length = 23)

yea I don't think it's necessary to introduce that extra code path just to keep per-column options.